## 파일 DataFrame 로딩, 기본 API

In [36]:
import pandas as pd

In [11]:
titanic_df = pd.read_csv(r'train.csv')

titanic_df.head(5)

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S

In [14]:
print("titanic 변수 type :",type(titanic_df))
print("DataFrame 크기 :",titanic_df.shape)

titanic 변수 type : <class 'pandas.core.frame.DataFrame'>
DataFrame 크기 : (891, 12)


In [15]:
titanic_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
PassengerId    891 non-null int64
Survived       891 non-null int64
Pclass         891 non-null int64
Name           891 non-null object
Sex            891 non-null object
Age            714 non-null float64
SibSp          891 non-null int64
Parch          891 non-null int64
Ticket         891 non-null object
Fare           891 non-null float64
Cabin          204 non-null object
Embarked       889 non-null object
dtypes: float64(2), int64(5), object(5)
memory usage: 83.6+ KB


* 전체 row 891개  
* 칼럼 타입, non-Null 값 갯수  
* 2개 칼럼이 float64, 5개 칼럼이 int64, 5개 칼럼이 object 타입  

In [17]:
titanic_df.describe()

PassengerId    Survived      Pclass         Age       SibSp  \
count   891.000000  891.000000  891.000000  714.000000  891.000000   
mean    446.000000    0.383838    2.308642   29.699118    0.523008   
std     257.353842    0.486592    0.836071   14.526497    1.102743   
min       1.000000    0.000000    1.000000    0.420000    0.000000   
25%     223.500000    0.000000    2.000000   20.125000    0.000000   
50%     446.000000    0.000000    3.000000   28.000000    0.000000   
75%     668.500000    1.000000    3.000000   38.000000    1.000000   
max     891.000000    1.000000    3.000000   80.000000    8.000000   

            Parch        Fare  
count  891.000000  891.000000  
mean     0.381594   32.204208  
std      0.806057   49.693429  
min      0.000000    0.000000  
25%      0.000000    7.910400  
50%      0.000000   14.454200  
75%      0.000000   31.000000  
max      6.000000  512.329200

In [18]:
# 해당 칼럼값의 유형과 건수 확인
value_counts = titanic_df['Pclass'].value_counts()
print(value_counts)

3    491
1    216
2    184
Name: Pclass, dtype: int64


In [20]:
# 해당 칼럼에 해당하는 Series 객체 반환
titanic_pclass = titanic_df['Pclass']
print(type(titanic_pclass))

<class 'pandas.core.series.Series'>


In [21]:
# 반환된 Series 객체 구성
titanic_pclass.head()

0    3
1    1
2    3
3    1
4    3
Name: Pclass, dtype: int64

* Series의 해당 칼럼 데이터 값 3의 index = 0

## 데이터 셀렉션 및 필터링

In [24]:
print('단일 칼럼 데이터 추출 :\n', titanic_df['Pclass'].head(3))

단일 칼럼 데이터 추출 :
 0    3
1    1
2    3
Name: Pclass, dtype: int64


In [27]:
print('여러 칼럼의 데이터 추출 :\n', titanic_df[['Survived','Pclass']].head(3))

여러 칼럼의 데이터 추출 :
    Survived  Pclass
0         0       3
1         1       1
2         1       3


In [33]:
# [ ] 안에 숫자 index는 KeyError 오류 발생

# 슬라이싱으로 해결
titanic_df[0:2]

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   

   Parch     Ticket     Fare Cabin Embarked  
0      0  A/5 21171   7.2500   NaN        S  
1      0   PC 17599  71.2833   C85        C

In [37]:
# ix[] 연산자
print('칼럼 위치 기반 인덱싱 데이터 추출 :', titanic_df.ix[0,2])
print('칼럼 명 기반 인덱싱 데이터 추출 :', titanic_df.ix[0,'Pclass'])

칼럼 위치 기반 인덱싱 데이터 추출 : 3
칼럼 명 기반 인덱싱 데이터 추출 : 3


C:\ProgramData\Anaconda3\lib\site-packages\ipykernel_launcher.py:2: DeprecationWarning: 
.ix is deprecated. Please use
.loc for label based indexing or
.iloc for positional indexing

See the documentation here:
http://pandas.pydata.org/pandas-docs/stable/indexing.html#ix-indexer-is-deprecated
  
C:\ProgramData\Anaconda3\lib\site-packages\ipykernel_launcher.py:3: DeprecationWarning: 
.ix is deprecated. Please use
.loc for label based indexing or
.iloc for positional indexing

See the documentation here:
http://pandas.pydata.org/pandas-docs/stable/indexing.html#ix-indexer-is-deprecated
  This is separate from the ipykernel package so we can avoid doing imports until


In [40]:
data = {'Name' : ['Chulmin', 'Eunkyung', 'Jinwoong', 'Soobeom'],
        'Year' : [2011, 2016, 2015, 2015],
        'Gender' : ['Male', 'Female', 'Male', 'Male']}

data_df = pd.DataFrame(data,index=['one','two','three','four'])
data_df

Name  Year  Gender
one     Chulmin  2011    Male
two    Eunkyung  2016  Female
three  Jinwoong  2015    Male
four    Soobeom  2015    Male

In [46]:
# reset_index()로 새로운 숫자형 인덱스 생성
data_df_reset = data_df.reset_index()
data_df_reset = data_df_reset.rename(columns={'index':'old_index'})

# 인덱스값에 1을 더해서 새로운 인덱스값 생성
data_df_reset.index = data_df_reset.index + 1

In [47]:
# iloc[] 연산자

print(data_df.iloc[0,0])
print(data_df_reset.iloc[0,1])

Chulmin
Chulmin


In [49]:
# loc[] 연산자

print(data_df.loc['one','Name'])
print(data_df_reset.loc[1,'Name'])

Chulmin
Chulmin


In [53]:
print('위치 기반 iloc slicing \n', data_df.iloc[0:1,0])
print('\n명칭 기반 loc slicing \n', data_df.loc['one':'two','Name'])

위치 기반 iloc slicing 
 one    Chulmin
Name: Name, dtype: object

명칭 기반 loc slicing 
 one     Chulmin
two    Eunkyung
Name: Name, dtype: object
